modules/document_repository/ajax/documentEditUpload.php

```diff
@@ -121,9 +125,13 @@
         $comments   = $_POST['commentsEdit'];
         $version    = $_POST['versionEdit'];
 
-        if (empty($category) && $category !== '0') {
-            header("HTTP/1.1 400 Bad Request");
-            exit;
+        // $category is a string representation of an ID, and so should be at
+        // least equal to zero
+        if (intval($category) < 0) {
```

Comment:
why does this need to be checked a second time? it appears above

Reply:
Once for `edit` actions, another for `upload` actions. They are in different branches of the code.